### PR TITLE
feat(grid3d): Discrete property value in grid readouts

### PR DESCRIPTION
**Pull request by LeonidPolukhin**
1) Grid3dLayer API takes an array of discrete property value-name pairs.
2) The array is used to display both discrete property names and values in readouts.
3) A grid story updated.
4) minimal npm version is bumped a little because of eslint-plugin requirements.

### CODE REVIEW

**Review by t0oF-azpn — APPROVED** [review]

File: typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ import fsShader from "./fragment.fs.glsl";
 import vsLineShader from "./vertex_lines.glsl";
 import fsLineShader from "./fragment_lines.glsl";
 import { localPhongLighting, utilities } from "../shader_modules";
-import { TGrid3DColoringMode } from "./grid3dLayer";
+import { IDiscretePropertyValueName, TGrid3DColoringMode } from "./grid3dLayer";
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts:25]: There is a warning here: `Imports "IDiscretePropertyValueName" are only used as type`

File: typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts [hunk 2]
```diff
@@ -147,10 +152,10 @@ export interface Grid3DLayerProps extends ExtendedLayerProps {
     propertiesData: string | number[] | Float32Array | Uint16Array;
 
     /**
-     * Discrete propety names to be displayed in cursor readouts.
-     * The property values are used as property name indices.
+     * Discrete propety value-name pairs to be displayed in cursor readouts.
+     * The property values are used as the array indices.
      */
-    discretePropertyNames?: string[];
+    discretePropertyValueNames?: IDiscretePropertyValueName[];
```
Comment (hkfb) [inline typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts:158]: Normally this is a breaking change, but maybe no-one else is using this layer?

File: typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts (hunk 1 above)
Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/layers/grid3d/privateGrid3dLayer.ts]: Fixed. IDiscretePropertyValueName is imported as type

File: typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts (hunk 2 above)
Comment (LeonidPolukhin) [reply typescript/packages/subsurface-viewer/src/layers/grid3d/grid3dLayer.ts:158]: Only Viz uses the prop as far as I know

**Pull request merged**